app.js

```diff
@@ -0,0 +1,764 @@
+const dotenv = require("dotenv");
+const axios = require("axios");
+const fs = require("fs");
+
+// import consts
+const { new_user } = require("./settings/new_user");
+
+// import modal blocks
+const modals = require("./blocks/modals");
+const {
+  launch_modal,
+  approver_details,
+  quote_line_details,
+  deal_stats,
+} = modals;
+
+// import message blocks
+const {
+  app_home,
+  channel_exists,
+  channel_created,
+  discount_approved,
+  discount_mention,
+  channel_message,
+  redirect_home,
+  thread_approved,
+  thread_ask,
+  thread_error,
+} = require("./blocks/messages");
+
+// import database functions
+const {
+  storeTokens,
+  storeLocalInstallation,
+  retrieveTokens,
+  fetchLocalInstallation,
+} = require("./db/helpers");
+
+// initialize env variables
+dotenv.config();
+
+// intialize app
+const app = new App({
+  signingSecret: process.env.SLACK_SIGNING_SECRET,
+  clientId: process.env.SLACK_CLIENT_ID,
+  clientSecret: process.env.SLACK_CLIENT_SECRET,
+  stateSecret: "apples-fans-boats",
+  scopes: [
+    "channels:history",
+    "channels:manage",
+    "chat:write",
+    "chat:write.public",
+    "commands",
+    "users:read",
+    "channels:read",
+    "links:read",
+  ],
+  installerOptions: {
+    userScopes: ["channels:history"],
+  },
+  installationStore: {
+    storeInstallation: async (installation) => {
+      await storeTokens(installation);
+      return storeLocalInstallation(installation);
+    },
+    fetchInstallation: async ({ teamId }) => {
+      const tokens = await retrieveTokens(teamId);
+      const installation = fetchLocalInstallation(tokens, teamId);
+      return installation;
+    },
+  },
+});
+
+/* UTILITY FUNCTIONS 🧰 */
+
+// check if approver users exist at each level
+const is_approvers_configured = (approver_users) => {
+  if (!approver_users) return false;
+  const { l1_user, l2_user, sales_ops_user, legal_user } = approver_users;
+  if (!l1_user || !l2_user || !sales_ops_user || !legal_user) {
+    return false;
+  }
+  return true;
+};
+
+// delay wrapper function
+const wait = async (ms) => {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms);
+  });
+};
+
+// send approval in thread with 3 second delay
+const sendApprovedMessage = async ({
+  token,
+  channel,
+  ts,
+  approver,
+  approval_level,
+  status,
+  user,
+  user_settings_obj,
+}) => {
+  try {
+    await wait(3000);
+
+    // post approval message
+    await app.client.chat.postMessage({
+      token: token,
+      channel: channel,
+      thread_ts: ts,
+      blocks: thread_approved({ approver, approval_level }),
+    });
+
+    // update status of original message
+    await app.client.chat.update({
+      token: token,
+      channel: channel,
+      ts: ts,
+      blocks: channel_message({
+        user,
+        companyName,
+        justification,
+        discount,
+        status,
+        user_settings_obj,
+      }),
+    });
+  } catch (error) {
+    console.error(error);
```

Comment:
Should we be sending the app an error message here? And for the rest of the actions as well

Reply:
I was just following the methodology they used in the [docs](https://slack.dev/bolt-js/concepts#shortcuts) but I'm happy to change this if you think there's a more suitable way to manage errors.

Do you think we need to add a `console.log` statement too?

Reply:
One thing we could do here is use the `app.error(() => {})` method that will handle all errors that appears ([link to this part of the bolt docs](https://slack.dev/bolt-js/concepts#error-handling))

However  I think this doesn't mean you can get rid of the `try/catch` statement here. But you can add your message for all errors in the `app.error()` method (I have not tried this yet so not sure if it works)

Reply:
@clrmachado, that's a great suggestion! @georgefairbairn, let me know if you're able to get that method to work. It's not super high priority but would be nice to have a more detailed error message